octoprint_mrbeam/service/profile/profile.py

```diff
@@ -0,0 +1,482 @@
+            self.save(self._default, allow_overwrite=True, make_default=True)
+
+    def select(self, identifier):
+        if identifier is None or not self.exists(identifier):
+            self._current = self.get_default()
+            return False
+        else:
+            self._current = self.get(identifier)
+            if self._current is None:
+                self._logger.error("Profile {} is invalid, cannot select, falling back to default".format(identifier))
+                self._current = self.get_default()
+                return False
+            else:
+                return True
+
+    def deselect(self):
+        self._current = None
+
+    def get_all(self):
+        return self._load_all()
+
+    def get(self, identifier):
+        try:
+            if self.exists(identifier):
+                return self._load_from_path(self._get_profile_path(identifier))
+            else:
+                return None
+        except InvalidProfileError:
+            return None
+
+    def remove(self, identifier):
+        if self._current is not None and self._current["id"] == identifier:
+            return False
+        elif settings().get(["profiles", self._id, "default"]) == identifier:
+            return False
+        return self._remove_from_path(self._get_profile_path(identifier))
+
+    def save(self, profile, allow_overwrite=False, make_default=False):
+        if "id" in profile:
+            identifier = profile["id"]
+        elif "name" in profile:
+            identifier = profile["name"]
+        else:
+            raise InvalidProfileError("profile must contain either id or name")
+
+        identifier = self._sanitize(identifier)
+        profile["id"] = identifier
+
+        self._migrate_profile(profile)
+        profile = dict_sanitize(profile, self._default)
+        profile = dict_merge(self._default, profile)
+
+        self._save_to_path(self._get_profile_path(identifier), profile, allow_overwrite=allow_overwrite)
+
+        if make_default:
+            settings().set(["profiles", self._id, "default"], identifier, force=True)
+            settings().save()
+
+        if self._current is not None and self._current["id"] == identifier:
+            self.select(identifier)
+        return self.get(identifier)
+
+    def is_default_unmodified(self):
+        default = settings().get(["profiles", self._id, "default"])
+        return default is None or default == self.DEFAULT_PROFILE_ID or not self.exists(self.DEFAULT_PROFILE_ID)
+
+    @property
+    def profile_count(self):
+        return len(self._load_all_identifiers())
+
+    @property
+    def last_modified(self):
+        dates = [os.stat(self._folder).st_mtime]
+        dates += [entry.stat().st_mtime for entry in scandir(self._folder) if entry.name.endswith(".profile")]
+        return max(dates)
+
+    def get_default(self):
+        default = settings().get(["profiles", self._id, "default"])
+        if default is not None and self.exists(default):
```

Comment:
I think this could be simplified, as we already handle the None value inside of self.exists()
```suggestion
        if self.exists(default):
```

Reply:
Or even more, in self.get() we already check if it exists, so we wouldn't need this statement

Reply:
This is not my implementation but OctoPrint's. We are using their class as a parent class with minor changes to fir our purpose